internal/manager/run.go

```diff
@@ -313,6 +314,27 @@ func Run(
 		}
 		configTranslator.InjectLicenseGetter(agent)
 	}
+	// enable KongLicense controller if license synchornizition from Konnect is disabled.
+	if c.KongLicenseEnabled && !c.Konnect.LicenseSynchronizationEnabled {
+		setupLog.Info("Starting KongLicense controller")
+		licenseController := &configuration.KongV1Alpha1KongLicenseReconciler{
+			Client:           mgr.GetClient(),
+			Log:              ctrl.LoggerFrom(ctx).WithName("controllers").WithName("KongLicense"),
+			Scheme:           mgr.GetScheme(),
+			LicenseCache:     configuration.NewLicenseCache(),
+			CacheSyncTimeout: c.CacheSyncTimeout,
+			StatusQueue:      kubernetesStatusQueue,
+			ControllerName:   c.LeaderElectionID,
+		}
+		dynamicLicenseController := configuration.WrapKongLicenseReconcilerToDynamicCRDController(
+			ctx, mgr, licenseController,
+		)
+		err := dynamicLicenseController.SetupWithManager(mgr)
+		if err != nil {
+			return fmt.Errorf("failed to start KongLicense controller: %w", err)
+		}
+		configTranslator.InjectLicenseGetter(licenseController)
```

Comment:
Ah, I see - we do not enable the `KongLicense` controller when the Konnect license agent is enabled. I think it would be worth having this covered in some unit test so it's not messed up in future refactors. We could extract a helper setup function (e.g. `setupLicenseGetter`) that we could test in isolation and verify the logic responsible for building the controller or the konnect agent.

Reply:
Moved to a `setupLicenseGetter` function in `setup.go`.